### PR TITLE
Added code fix provider for SA1206 and tests.

### LINKED ISSUE #1003: Code fix: SA1206 (DeclarationKeywordsMustFollowOrder)
Implement a code fix for #66.

### PR DESCRIPTION
**Pull request by bexxx**
Fixes #1003

### CODE REVIEW

**Comment by bexxx** [comment]
Resolves #1003

File: StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/SA1206DeclarationKeywordsMustFollowOrder.cs [hunk 1]
```diff
@@ -111,38 +86,39 @@ private static void HandleDeclaration(SyntaxNodeAnalysisContext context)
             CheckModifiersOrderAndReportDiagnostics(context, modifiers);
         }
 
+        // TODO: Can this be merged with DeclarationModifiersHelper.GetModifiersFromDeclaration?
```
Comment (pdelvo) [inline StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/SA1206DeclarationKeywordsMustFollowOrder.cs]: I think `DeclarationModifiersHelper.GetModifiers` should work in places where this method is used.

File: StyleCop.Analyzers/StyleCop.Analyzers/Helpers/DeclarationModifiersHelper.cs [hunk 2]
```diff
@@ -100,5 +100,56 @@ internal static SyntaxTokenList GetModifiers(this MemberDeclarationSyntax syntax
 
             return default(SyntaxTokenList);
         }
+
+        internal static SyntaxNode WithModifiers(this SyntaxNode node, SyntaxTokenList modifiers)
+        {
+            switch (node.Kind())
+            {
+                case SyntaxKind.MethodDeclaration:
```
Comment (pdelvo) [inline StyleCop.Analyzers/StyleCop.Analyzers/Helpers/DeclarationModifiersHelper.cs]: Nit: The case labels should not be indented.

**Comment by pdelvo** [comment]
Sorry for reviewing this so late. I just found this PR when I looked through the old ones. I would really like to get this merged in. It looks like there is a merge conflict so if you still have interest in contributing it would be great if you could rebase this PR on master and resolve the conflict. If you have trouble with that I can help you with that.

**Comment by bexxx** [comment]
Oh, I missed your comments too :).
Sure thing, I'll update the branch to resolve the merge conflicts soon and ping you again.

**Comment by pdelvo** [comment]
Looks good to me :+1:

**Pull request merged**